examples/step-42/step-42.cc

```diff
@@ -876,8 +876,15 @@ namespace Step42
     output_dir = prm.get("output directory");
     if (output_dir != "" && *(output_dir.rbegin()) != '/')
       output_dir += "/";
-    const int ierr = mkdir(output_dir.c_str(), 0777);
-    AssertThrow(ierr == 0, ExcIO());
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        const int ierr = mkdir(output_dir.c_str(), 0777);
+        AssertThrow(ierr == 0, ExcIO());
+      }
+    const int ierr = MPI_Barrier(mpi_communicator);
```

Comment:
Why did you add the `MPI_Barrier`?

Reply:
I thought that I needed it but I don't recall why. I got rid of it.